src/writer/async_tokio.rs

```diff
@@ -82,6 +84,178 @@ impl<W: AsyncWrite + Unpin> Writer<W> {
+    ///
+    ///    writer
+    ///        .create_element("empty")
+    ///        .with_attribute(("attr1", "value1"))
+    ///        .with_attribute(("attr2", "value2"))
+    ///        .write_empty_async()
+    ///        .await
+    ///        .expect("failure");
+    ///
+    ///     tokio_buffer.flush().await.expect("flush failed");
+    ///
+    ///     assert_eq!(
+    ///      std::str::from_utf8(&buffer).unwrap(),
+    ///      r#"<empty attr1="value1" attr2="value2"/>"#
+    ///     );
+    /// }
+    pub async fn write_empty_async(self) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event_async(Event::Empty(self.start_tag))
+            .await?;
+        Ok(self.writer)
+    }
+
+    /// Create a new scope for writing XML inside the current element.
+    /// # Example
+    ///
+    /// ```rust
+    /// # use quick_xml::Result;
+    /// # use quick_xml::Writer;
+    /// # use quick_xml::events::BytesText;
+    /// # use tokio::io::AsyncWriteExt;
+    /// # #[tokio::main(flavor = "current_thread")] async fn main() {
+    ///    let mut buffer = Vec::new();
+    ///         let mut tokio_buffer = tokio::io::BufWriter::new(&mut buffer);
+    ///         let mut writer = Writer::new_with_indent(&mut tokio_buffer, b' ', 4);
+
+    ///         writer
+    ///             .create_element("outer")
+    ///             .with_attributes([("attr1", "value1"), ("attr2", "value2")])
+    ///             .write_inner_content_async(|writer| async move {
+    ///                 let fruits = ["apple", "orange", "banana"];
+    ///                 for (quant, item) in fruits.iter().enumerate() {
+    ///                     writer
+    ///                         .create_element("fruit")
+    ///                         .with_attributes([("quantity", quant.to_string().as_str())])
+    ///                         .write_text_content_async(BytesText::new(item))
+    ///                         .await?;
+    ///                 }
+    ///                 writer
+    ///                     .create_element("inner")
+    ///                     .write_inner_content_async(|writer| async move {
+    ///                         writer.create_element("empty").write_empty_async().await?;
+    ///                         Ok(writer)
+    ///                     })
+    ///                     .await?;
+
+    ///                 Ok(writer)
+    ///             })
+    ///             .await
+    ///             .expect("failure");
+
+    ///         tokio_buffer.flush().await.expect("flush failed");
+    ///         assert_eq!(
+    ///             std::str::from_utf8(&buffer).unwrap(),
+    ///             r#"<outer attr1="value1" attr2="value2">
+    ///     <fruit quantity="0">apple</fruit>
+    ///     <fruit quantity="1">orange</fruit>
+    ///     <fruit quantity="2">banana</fruit>
+    ///     <inner>
+    ///         <empty/>
+    ///     </inner>
+    /// </outer>"#
+    ///         );
+    /// }
```

Comment:
```suggestion
    /// # }
```
And also please dedent the visible code in example and check that indent is correct (the first visible line has different indent that the others)